app/controllers/fuel_controller.rb

```diff
@@ -14,4 +14,44 @@ def s3_bucket
     )
   end
 
+  def self.paginated_action(options={})
+    before_filter(options) do |controller|
+      # Set defaults
+      @pagination_current_page = 1
+      @pagination_per_page = nil
+
+      if request.headers['Range-Unit'] == 'items' && request.headers['Range'].present?
```

Comment:
How does these Range options interact with the existing `config.paginates_per` configuration?